### PR TITLE
sinkv2(ticdc): add kafka metrics

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

- Add kafka metrics

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sinkv2/eventsink/mq/dmlproducer/kafka_dml_producer.go [hunk 1]
```diff
@@ -62,6 +66,73 @@ type kafkaDMLProducer struct {
...
+
+	// Start collecting metrics.
+	go k.collector.Run(ctx)
+
+	go func() {
+		if err := k.run(ctx); err != nil && errors.Cause(err) != context.Canceled {
+			select {
+			case <-ctx.Done():
+				return
+			case errCh <- err:
```
Comment (nongfushanquan) [inline cdc/sinkv2/eventsink/mq/dmlproducer/kafka_dml_producer.go:124]: Pls add some err logs here

File: cdc/sinkv2/metrics/kafka/collector.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+	// metrics at broker level.
+	incomingByteRateMetricNamePrefix   = "incoming-byte-rate-for-broker-"
+	outgoingByteRateMetricNamePrefix   = "outgoing-byte-rate-for-broker-"
+	requestRateMetricNamePrefix        = "request-rate-for-broker-"
+	requestSizeMetricNamePrefix        = "request-size-for-broker-"
+	requestLatencyInMsMetricNamePrefix = "request-latency-in-ms-for-broker-"
+	requestsInFlightMetricNamePrefix   = "requests-in-flight-for-broker-"
+	responseRateMetricNamePrefix       = "response-rate-for-broker-"
+	responseSizeMetricNamePrefix       = "response-size-for-broker-"
```
Comment (zhaoxinyu) [inline cdc/sinkv2/metrics/kafka/collector.go:48]: We can add another metric to track the amount of topics created for each changefeed in TiCDC.

File: cdc/sinkv2/metrics/statistics.go [hunk 3]
```diff
@@ -125,6 +108,16 @@ func (b *Statistics) AddRowsCount(count int) {
 	atomic.AddUint64(&b.totalRows, uint64(count))
 }
 
+// SubRowsCount records total number of rows needs to flush
+func (b *Statistics) SubRowsCount(count int) {
```
Comment (zhaoxinyu) [inline cdc/sinkv2/metrics/statistics.go:112]: The function `SubRowsCount()` and `TotalRowsCount` are obsolete. Maybe we can delete them.

**Comment by zhaoxinyu** [comment]
Rest LGTM

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**